jwst/ramp_fitting/ramp_fit.py

```diff
@@ -616,6 +544,249 @@ def ols_ramp_fit(data, err, groupdq, inpixeldq, buffsize, save_opt, readnoise_2d
+    #                    sum_over_integs_and_segs(1/var_seg)
+    new_model, int_model, opt_model = ramp_fit_overall_slopes_and_variances(
+        n_int, nrows, ncols, imshape, orig_cubeshape, orig_nreads, buffsize,
+        groupdq, first_pass_ans, second_pass_ans, opt_res, save_opt,
+        int_times, instrume, tstart, nframes, nreads, groupgap, dropframes1)
+
+    # Unpack output to return.
+    if int_model is not None:
+        int_data = int_model.data.copy()
+        int_dq = int_model.dq.copy()
+        int_var_poisson = int_model.var_poisson.copy()
+        int_var_rnoise = int_model.var_rnoise.copy()
+        int_err = int_model.err.copy()
+        int_int_times = int_model.int_times.copy()
+    else:
+        int_data = None
+        int_dq = None
+        int_var_poisson = None
+        int_var_rnoise = None
+        int_err = None
+        int_int_times = None
+    if opt_model is not None:
+        opt_slope = opt_model.slope.copy()
+        opt_sigslope = opt_model.sigslope.copy()
+        opt_var_poisson = opt_model.var_poisson.copy()
+        opt_var_rnoise = opt_model.var_rnoise.copy()
+        opt_yint = opt_model.yint.copy()
+        opt_sigyint = opt_model.sigyint.copy()
+        opt_pedestal = opt_model.pedestal.copy()
+        opt_weights = opt_model.weights.copy()
+        opt_crmag = opt_model.crmag.copy()
+        actual_segments = opt_slope.shape[1]
+        actual_CRs = opt_crmag.shape[1]
+    else:
+        opt_slope = None
+        opt_sigslope = None
+        opt_var_poisson = None
+        opt_var_rnoise = None
+        opt_yint = None
+        opt_sigyint = None
+        opt_pedestal = None
+        opt_weights = None
+        opt_crmag = None
+        actual_segments = 0
+        actual_CRs = 0
+
+    return new_model.data, new_model.dq, new_model.var_poisson, new_model.var_rnoise, new_model.err, \
+            int_data, int_dq, int_var_poisson, int_var_rnoise, int_err, int_int_times, \
+            opt_slope, opt_sigslope, opt_var_poisson, opt_var_rnoise, opt_yint, opt_sigyint, \
+            opt_pedestal, opt_weights, opt_crmag, actual_segments, actual_CRs
+
+
+def ramp_fit_miri_multigroups(data, err, groupdq, cubeshape, imshape, nreads, ngroups):
+    """
+    For MIRI datasets having >1 group, if all pixels in the final group are
+    flagged as DO_NOT_USE, resize the input model arrays to exclude the
+    final group.  Similarly, if leading groups 1 though N have all pixels
+    flagged as DO_NOT_USE, those groups will be ignored by ramp fitting, and
+    the input model arrays will be resized appropriately. If all pixels in
+    all groups are flagged, return None for the models.
+
+    Parameters
+    ----------
+    data: 4-D image cube with dimensions (integrations, groups, rows, columns)
+    err: Error array
+    groupdq: GroupDQ array
+    cubeshape: Image cube dimensions per integration
+    imshape: Image shape per group
+    nreads: Number of reads
+    ngroups: Number of groups
```

Comment:
Could you make these docstrings more `numpy` standard?:

```
data : np.ndarray
    4-D image cube with dimensions (integrations, groups, rows, columns)
err : np.ndarray
```

etc

Reply:
Yes.  I will expand these to conform better to expected standards.